internal/aws/proxy/server.go

```diff
@@ -93,6 +93,13 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 			req.URL.Host = awsURL.Host
 			req.Host = awsURL.Host
 
+			// According to RFC7230
+			// A user agent SHOULD NOT send a Content-Length header field when the request message does not contain
+   			// a payload body and the method semantics do not anticipate such a body.
```

Comment:
so should this be contingent on the HTTP method used? For POST for example, setting "0" is valid.

Reply:
I tested this change with AWS OpenSearch and apparently on POST request also with Content length zero, I was getting Sigv4 issues

Reply:
i have changed the comment but the fix is valid